compiler/rustc_session/src/options.rs

```
@@ -1351,9 +1351,25 @@ pub mod parse {
         let mut v: Vec<&str> = v.split(",").collect();
         v.sort_unstable();
         for &val in v.iter() {
-            let variant = match val {
+            // Split each entry on '=' if it has an argument
+            let (key, arg) = match val.split_once('=') {
+                Some((k, a)) => (k, Some(a)),
+                None => (val, None),
+            };
+
+            let variant = match key {
                 "Enable" => AutoDiff::Enable,
                 "PrintTA" => AutoDiff::PrintTA,
+                "PrintTAFn" => {
+                    if let Some(fun) = arg {
+                        AutoDiff::PrintTAFn(fun.to_string())
+                    } else {
+                        eprintln!(
+                            "Missing argument for PrintTAFn (expected PrintTAFn=<function_name>)"
```

Did you see eprintln! usages somewhere in the frontend?
Usually rustc output goes through some DiagContext to make sure it's properly formatted, could be translated, potentially cached, and just generally handled in the expected way. I would be worried to add a new approach for emitting messages, given how big rustc is it likely would break something.

---

resolved

---

Do not attempt to set the policy for whether an error is ignored or fatal-erred or warned on here.